### PR TITLE
Add Spotless Fail Fast to the build process

### PR DESCRIPTION
**Pull request by 7suyash7**
#6588 
Add Spotless fail fast to the build process.

Output:
```
 ./gradlew build            ok  3s  19:34:06

> Configure project :
Generating project version as supplied is version not semver: unspecified

> Task :ethereum:core:spotlessJavaCheck FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':ethereum:core:spotlessJavaCheck'.
> The following files had format violations:
      src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java
          @@ -337,15 +337,15 @@
           ······return·Optional.empty();
           ····}

          -········transactionBuilder.guessType();
          +····transactionBuilder.guessType();
           ····if·(transactionBuilder.getTransactionType().supportsBlob())·{
           ······transactionBuilder.maxFeePerBlobGas(maxFeePerBlobGas);
           ····}
           ····if·(transactionBuilder.getTransactionType().requiresChainId())·{
          -····transactionBuilder.chainId(
          +······transactionBuilder.chainId(
           ··········protocolSchedule
          -······.getChainId()
          -······.orElse(BigInteger.ONE));·//·needed·to·make·some·transactions·valid
          +··············.getChainId()
          +··············.orElse(BigInteger.ONE));·//·needed·to·make·some·transactions·valid
           ····}

           ····final·Transaction·transaction·=·transactionBuilder.build();
  Run './gradlew :ethereum:core:spotlessApply' to fix these violations.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.8/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 15s
253 actionable tasks: 4 executed, 249 up-to-date
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -432,8 +436,16 @@ allprojects {
...
   }
+
+  plugins.withType(JavaPlugin) {
+    tasks.withType(JavaCompile) {
+      it.dependsOn(rootProject.tasks.named('spotlessCheckFast'))
+    }
+  }
 }
 
+check.dependsOn spotlessCheckFast
```
Comment (usmansaleem) [inline build.gradle:447]: can you move it line number 470 i.e. `check.dependsOn('checkPluginAPIChanges', 'checkMavenCoordinateCollisions', 'spotlessCheckFast')` ?

**Review by usmansaleem — APPROVED** [review]
One minor change, otherwise looks good to me.

**Comment by 7suyash7** [comment]
> One minor change, otherwise looks good to me.

Done

**Comment by fab-10** [comment]
Can we have the spotless check enabled only when calling `./gradlew build`? Because now it runs every compilation tasks, for example also when making a simple code change and running a test, this add a delay and it is also a bit useless to do spotless check when just commenting some lines or testing a quick change

**Pull request merged**